### ISSUE TITLE
Building from head (135b6aa) seems broken

### ISSUE THREAD

**Issue by ccaviness** [CleverRaven/Cataclysm-DDA#63858]
### Describe the bug

Using homebrew on an Apple Silicon MacBook Pro, it failed. Tried again with a local git repo, and failed with the same error.

```
src/mission.h:138:20: error: use of undeclared identifier 'int_or_var'
    cata::optional<int_or_var<T>> reveal_radius;
                   ^
src/mission.h:140:5: error: no template named 'int_or_var'
    int_or_var<T> min_distance;
    ^
src/mission.h:146:5: error: no template named 'int_or_var'
    int_or_var<T> search_range;
    ^
src/mission.h:147:20: error: use of undeclared identifier 'int_or_var'
    cata::optional<int_or_var<T>> z;
                   ^
```

### Attach save file

N/A

### Steps to reproduce

1.  `git clone https://github.com/CleverRaven/Cataclysm-DDA`
2.  `cd Cataclysm-DDA`
3.  `make`

### Expected behavior

Compiles successfully.

### Versions and configuration

macOS 13.3 on an M1 MacBook Pro

### Additional context

```
$ git clone --depth=1 https://github.com/CleverRaven/Cataclysm-DDA
Cloning into 'Cataclysm-DDA'...
remote: Enumerating objects: 8444, done.
remote: Counting objects: 100% (8444/8444), done.
remote: Compressing objects: 100% (7507/7507), done.
remote: Total 8444 (delta 1514), reused 4122 (delta 795), pack-reused 0
Receiving objects: 100% (8444/8444), 105.18 MiB | 22.75 MiB/s, done.
Resolving deltas: 100% (1514/1514), done.
Updating files: 100% (7795/7795), done.
$ cd Cataclysm-DDA/
$ make
// NOLINT(cata-header-guard)
#define VERSION "135b6aa"
clang++ -Isrc -isystem src/third-party -DMACOSX -DGIT_VERSION -DBACKTRACE -DLOCALIZE -Og -Werror -Wall -Wextra -Wformat-signedness -Wlogical-op -Wmissing-declarations -Wmissing-noreturn -Wnon-virtual-dtor -Wold-style-cast -Woverloaded-virtual -Wpedantic -Wsuggest-override -Wunused-macros -Wzero-as-null-pointer-constant -Wno-unknown-warning-option -Wredundant-decls  -g  -fsigned-char -stdlib=libc++ -std=c++14 -mmacosx-version-min=13.3  -MMD -MP -Wno-error -c pch/main-pch.hpp -o pch/main-pch.hpp.pch
clang++ -Isrc -isystem src/third-party -DMACOSX -DGIT_VERSION -DBACKTRACE -DLOCALIZE -Og -Werror -Wall -Wextra -Wformat-signedness -Wlogical-op -Wmissing-declarations -Wmissing-noreturn -Wnon-virtual-dtor -Wold-style-cast -Woverloaded-virtual -Wpedantic -Wsuggest-override -Wunused-macros -Wzero-as-null-pointer-constant -Wno-unknown-warning-option -Wredundant-decls  -g  -fsigned-char -stdlib=libc++ -std=c++14 -mmacosx-version-min=13.3  -MMD -MP -Winvalid-pch -include-pch pch/main-pch.hpp.pch -c src/achievement.cpp -o obj/achievement.o
clang++ -Isrc -isystem src/third-party -DMACOSX -DGIT_VERSION -DBACKTRACE -DLOCALIZE -Og -Werror -Wall -Wextra -Wformat-signedness -Wlogical-op -Wmissing-declarations -Wmissing-noreturn -Wnon-virtual-dtor -Wold-style-cast -Woverloaded-virtual -Wpedantic -Wsuggest-override -Wunused-macros -Wzero-as-null-pointer-constant -Wno-unknown-warning-option -Wredundant-decls  -g  -fsigned-char -stdlib=libc++ -std=c++14 -mmacosx-version-min=13.3  -MMD -MP -Winvalid-pch -include-pch pch/main-pch.hpp.pch -c src/action.cpp -o obj/action.o
clang++ -Isrc -isystem src/third-party -DMACOSX -DGIT_VERSION -DBACKTRACE -DLOCALIZE -Og -Werror -Wall -Wextra -Wformat-signedness -Wlogical-op -Wmissing-declarations -Wmissing-noreturn -Wnon-virtual-dtor -Wold-style-cast -Woverloaded-virtual -Wpedantic -Wsuggest-override -Wunused-macros -Wzero-as-null-pointer-constant -Wno-unknown-warning-option -Wredundant-decls  -g  -fsigned-char -stdlib=libc++ -std=c++14 -mmacosx-version-min=13.3  -MMD -MP -Winvalid-pch -include-pch pch/main-pch.hpp.pch -c src/active_item_cache.cpp -o obj/active_item_cache.o
clang++ -Isrc -isystem src/third-party -DMACOSX -DGIT_VERSION -DBACKTRACE -DLOCALIZE -Og -Werror -Wall -Wextra -Wformat-signedness -Wlogical-op -Wmissing-declarations -Wmissing-noreturn -Wnon-virtual-dtor -Wold-style-cast -Woverloaded-virtual -Wpedantic -Wsuggest-override -Wunused-macros -Wzero-as-null-pointer-constant -Wno-unknown-warning-option -Wredundant-decls  -g  -fsigned-char -stdlib=libc++ -std=c++14 -mmacosx-version-min=13.3  -MMD -MP -Winvalid-pch -include-pch pch/main-pch.hpp.pch -c src/activity_actor.cpp -o obj/activity_actor.o
In file included from src/activity_actor.cpp:54:
In file included from src/martialarts.h:13:
In file included from src/effect_on_condition.h:9:
In file included from src/condition.h:12:
src/mission.h:138:20: error: use of undeclared identifier 'int_or_var'
    cata::optional<int_or_var<T>> reveal_radius;
                   ^
src/mission.h:140:5: error: no template named 'int_or_var'
    int_or_var<T> min_distance;
    ^
src/mission.h:146:5: error: no template named 'int_or_var'
    int_or_var<T> search_range;
    ^
src/mission.h:147:20: error: use of undeclared identifier 'int_or_var'
    cata::optional<int_or_var<T>> z;
                   ^
4 errors generated.
make: *** [obj/activity_actor.o] Error 1
crc-macbookpro3:Cataclysm-DDA crc$ git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
```

**Comment by ccaviness** [comment]
I think #63251 is the culprit.

**Comment by ccaviness** [comment]
Ah, pending pull request #63856 should fix this.

**Comment by pjf** [comment]
I can confirm this, and also confirm that applying #63856 resolved it locally.

**Comment by ccaviness** [comment]
Fixed by #63856
